File: GeneratedEntityReaderClasses/Components/LightComponent.cs
namespace GeneratedNoitaClasses;
public class LightComponent : NoitaComponentBase,  iNoitaType<LightComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool update_properties { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaUInt r { get; set; } = new();
	public NoitaUInt g { get; set; } = new();
	public NoitaUInt b { get; set; } = new();
	public NoitaFloat offset_x { get; set; } = new();
	public NoitaFloat offset_y { get; set; } = new();
	public NoitaFloat fade_out_time { get; set; } = new();
	public NoitaFloat blinking_freq { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		update_properties.Read(s);
		radius.Read(s);
		r.Read(s);
		g.Read(s);
		b.Read(s);
		offset_x.Read(s);
		offset_y.Read(s);
		fade_out_time.Read(s);
		blinking_freq.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		update_properties.Write(s);
		radius.Write(s);
		r.Write(s);
		g.Write(s);
		b.Write(s);
		offset_x.Write(s);
		offset_y.Write(s);
		fade_out_time.Write(s);
		blinking_freq.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/LightComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaUInt

from NoitaStream import NoitaStream
@dataclass
class LightComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    update_properties:NoitaBool = field(default_factory=NoitaBool)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    r:NoitaUInt = field(default_factory=NoitaUInt)
    g:NoitaUInt = field(default_factory=NoitaUInt)
    b:NoitaUInt = field(default_factory=NoitaUInt)
    offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    fade_out_time:NoitaFloat = field(default_factory=NoitaFloat)
    blinking_freq:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.update_properties.Read(s)
        self.radius.Read(s)
        self.r.Read(s)
        self.g.Read(s)
        self.b.Read(s)
        self.offset_x.Read(s)
        self.offset_y.Read(s)
        self.fade_out_time.Read(s)
        self.blinking_freq.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.update_properties.Write(s)
        self.radius.Write(s)
        self.r.Write(s)
        self.g.Write(s)
        self.b.Write(s)
        self.offset_x.Write(s)
        self.offset_y.Write(s)
        self.fade_out_time.Write(s)
        self.blinking_freq.Write(s)

